business/mesh.go

```diff
@@ -474,54 +470,70 @@ func (in *MeshService) resolveRemoteClustersFromSecrets() ([]Cluster, error) {
 	// Strictly speaking, this list may be incomplete: it's list of visible clusters for a control plane.
 	// But, for now, let's use it as the absolute "list of clusters in the mesh (excluding home cluster)".
 
-	// "Remote secrets" are created using the command `istioctl x create-remote-secret` which
-	// labels the secrets with istio/multiCluster=true. Let's use that label to fetch the secrets of interest.
-	secrets, err := in.k8s.GetSecrets(conf.IstioNamespace, "istio/multiCluster=true")
-	if err != nil {
-		if errors.IsForbidden(err) {
-			// A forbidden error means that we don't have privileges to list secrets in the Istio namespace.
-			// This may be because sysadmin may not want us to do that for security and probably
-			// because it is known that the environment is a single-cluster. So, return
-			// and empty list of clusters, avoid the warning error and use a trace log message.
-			log.Trace("Not enough privileges to list secrets with istio/multiCluster=true label.")
-			return []Cluster{}, nil
+	// Remote cluster secrets are mounted on the file system by the Kiali installer under the "/kiali-remote-cluster-secrets" directory.
+	// Each cluster has its own subdirectory, usually named the same as the cluster (though not necessarily). e.g. "/kiali-remote-cluster-secrets/<cluster name>"
+	// Cluster configs are found in a file whose name is the cluster name; e.g. "/kiali-remote-cluster-secrets/<cluster name>/<cluster name>"
+	// So walk the directory "/kiali-remote-cluster-secrets" to obtain all known remote cluster configs.
+
+	// Defines where the files are located that contain the remote cluster secrets
+	const remoteClusterSecretsDir = "/kiali-remote-cluster-secrets"
+	remoteClusterSecrets := make(map[string][]byte, 0)
+
+	secretDirs, err := os.ReadDir(remoteClusterSecretsDir)
+	if err == nil {
```

Comment:
You can reverse the `err` check to `err != nil` and return early. You may end up with more lines of code but it'll be much more readable than a deeply nested `if/else` tree and I'd prioritize readability.

Reply:
> You can reverse the err check to err != nil and return early.

We don't return after that if-stmt. It keeps going to do more processing, even if there are no secrets (or we fail to read secrets). So there really isn't a true short-circuit we could do here.

Reply:
> So there really isn't a true short-circuit we could do here

Can you use `continue` instead of `return` like you are already doing on line 493.
```
for _, sd := range secretDirs {
	secretAbsDir := remoteClusterSecretsDir + "/" + sd.Name()
	secretFiles, err := os.ReadDir(secretAbsDir)
	if err != nil {
            log.(...)
            continue
       }
       for ... {
```

The outermost I think can return empty remote secrets?
```
	secretDirs, err := os.ReadDir(remoteClusterSecretsDir)
	if err == nil {
            log.Error(...)
            return []Cluster{}, nil
        }
```

This might just be a personal preference but it's just hard for me to read what's going on when it's nested like this. Sometimes it can't be helped.